electrum/interface.py

```diff
@@ -1560,12 +1560,13 @@ async def get_donation_address(self) -> str:
         # check response
         if not res:  # ignore empty string
             return ''
-        if not bitcoin.is_address(res):
+        address = res.removeprefix('bitcoin:')
+        if not bitcoin.is_address(address):
```

Comment:
If `res` was not a str, previously `bitcoin.is_address()` handled that gracefully and returned False.
Now `.removeprefix()` would raise `AttributeError`.
Let's just check explicitly for type and raise the proper thing:
```suggestion
        if not isinstance(res, str):
            raise RequestCorrupted(f'{res!r} should be a str')
        address = res.removeprefix('bitcoin:')
        if not bitcoin.is_address(address):
```